src/main/java/com/zufar/onlinestore/cart/entity/ShoppingSession.java

```diff
@@ -0,0 +1,84 @@
+package com.zufar.onlinestore.cart.entity;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.UUID;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "shopping_session")
+public class ShoppingSession {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+
+    @Column(name = "user_id", nullable = false)
+    private UUID userId;
+
+    @OneToMany(mappedBy = "shoppingSession", cascade = CascadeType.ALL)
+    private Collection<ShoppingSessionItem> items;
+
+    @Column(name = "items_quantity", nullable = false)
+    private Integer itemsQuantity;
+
+    @Column(name = "products_quantity", nullable = false)
+    private Integer productsQuantity;
+
+    @Column(name = "created_at", nullable = false)
+    private LocalDateTime createdAt;
+
+    @Column(name = "closed_at", nullable = false)
+    private LocalDateTime closedAt;
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object)
+            return true;
+        if (object == null || getClass() != object.getClass())
+            return false;
+        ShoppingSession that = (ShoppingSession) object;
+        return Objects.equals(itemsQuantity, that.itemsQuantity) &&
+                Objects.equals(productsQuantity, that.productsQuantity) &&
+                Objects.equals(id, that.id) &&
+                Objects.equals(userId, that.userId) &&
+                Objects.equals(createdAt, that.createdAt) &&
+                Objects.equals(closedAt, that.closedAt) &&
+                Objects.equals(items, that.items);
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
For hashcode there are 2 options::

     1) if it is a surrogate id that is always generated manually (that is, we know it before saving it to the database) or it is a business id that is also known in advance and BEFORE saving and AFTER saving the id is the same:

     @Override
     public int hashCode() {
             return Objects.hashCode(id);
     }

    2) if it is a surrogate id that is always generated by the database (at the time of saving), and therefore BEFORE saving it is null, and AFTER it acquires a non null value:

     @Override
     public int hashCode() {
             return getClass().hashCode();
     }

Reply:
No, using getClass().hashCode() as the implementation for the hashCode() method is not recommended and should be avoided. This approach would produce the same hash code for all instances of the same class, which is incorrect and defeats the purpose of generating unique hash codes for different objects.

A correct implementation of hashCode() should take into account the relevant fields of the object that define its state. In the case of entities, using the id field (which represents the primary key and is typically unique for each entity) is a common and recommended approach for generating a hash code. It ensures that objects with different identities produce different hash codes.